task_processing/plugins/mesos/timeout_executor.py

```diff
@@ -0,0 +1,109 @@
+import collections
+import logging
+import time
+from threading import Lock
+from threading import Thread
+
+from six.moves.queue import Queue
+
+from task_processing.interfaces.task_executor import TaskExecutor
+
+log = logging.getLogger(__name__)
+
+TaskEntry = collections.namedtuple('TaskEntry', ['task_id', 'deadline'])
+
+
+class TimeoutExecutor(TaskExecutor):
+    def __init__(self, downstream_executor):
+        self.downstream_executor = downstream_executor
+
+        self.tasks_lock = Lock()
+        # Tasks that are pending termination
+        self.killed_tasks = []
```

Comment:
not sure i understand the purpose of this collection. in the code we walk running_tasks to invoke downstream_executor.kill, i don't see killed_tasks entries used for anything?

Reply:
Having killed_tasks or not won't affect functionality. It is for tracking/debug purposes. A terminal event is expected for a task being killed.